docs/advanced_usage/advanced_tests.rst

```diff
@@ -0,0 +1,432 @@
+
+        def test_run(self):
+            com_aux.send_message(STIMULUS)
+            resp = com_aux.receive_message()
+            self.assertEqual(resp, RESPONSE)
+
+
+We can use the configured and instantiated auxiliary ``com_aux`` (imported by it's alias) in the test directly.
+
+Implementation of Advanced Tests - Custom Setup
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you need to have more complex tests, you can do the following:
+
+-  ``BasicTest`` is a specific implementation of ``unittest.TestCase``
+   therefore it contains 3 steps/methods **setUp()**, **tearDown()** and
+   **test_run()** that can be overwritten.
+-  ``BasicTest`` will contain the list of **auxiliaries** you can use.
+   It will be hold in the attribute ``test_auxiliary_list``.
+-  ``BasicTest`` also contains the following information
+   ``test_section_id``, ``test_suite_id``, ``test_case_id``.
+-  Import *logging* or/and *message* (if needed) to communicate with the
+   **auxiliary**(in that case use RemoteTest instead of BasicTest)
+
+**test_suite_2.py**:
+
+.. code:: python
+
+    """
+    I want to run the following tests documented in the following test-specs <TEST_CASE_SPECS>.
+    """
+    import pykiso
+    from pykiso import message
+    from pykiso.auxiliaries import aux1
+
+
+    @pykiso.define_test_parameters(suite_id=2, case_id=1, aux_list=[aux1])
+    class MyTest(pykiso.BasicTest):
+        def setUp(self):
+           # I loop through all the auxiliaries
+           for aux in self.test_auxiliary_list:
+               if aux.name == "aux1": # If I find the auxiliary to which I need to send a special message, I compose the message and send it.
+                   # Compose the message to send with some additional information
+                   tlv = { TEST_REPORT:"Give me something" }
+                   testcase_setup_special_message = message.Message(msg_type=message.MessageType.COMMAND, sub_type=message.MessageCommandType.TEST_CASE_SETUP,
+                                                           test_section=self.test_section_id, test_suite=self.test_suite_id, test_case=self.test_case_id, tlv_dict=tlv)
+                   # Send the message
+                   aux.run_command(testcase_setup_special_message, blocking=True, timeout_in_s=10)
+               else: # Do not forget to send a setup message to the other auxiliaries!
+                   # Compose the normal message
+                   testcase_setup_basic_message = message.Message(msg_type=message.MessageType.COMMAND, sub_type=message.MessageCommandType.TEST_CASE_SETUP,
+                                                           test_section=self.test_section_id, test_suite=self.test_suite_id, test_case=self.test_case_id)
+                   # Send the message
+                   aux.run_command(testcase_setup_basic_message, blocking=True, timeout_in_s=10)
+
+.. _implementation-of-the-tests-advance-1:
+
+Implementation of Advanced Tests - Test Templates
```

Comment:
We should also at some point explain the general test flow, e.g.: 
```
SuiteSetup().test_suite_setUp 
TestCase1.setUpClass
TestCase1().setUp
TestCase1().test_run
TestCase1().tearDown
TestCase1().setUp
TestCase1().test_run_2
TestCase1().tearDown
TestCase1.tearDownClass
TestCase2.setUpClass
TestCase2().setUp
TestCase2().test_run
TestCase2().tearDown
TestCase2().setUp
TestCase2().test_run_2
TestCase2().tearDown
TestCase2.tearDownClass
TestSuite().test_suite_tearDown
```

Reply:
I added it in basic_tests.rst (the end)